framework-plugins/lisk-framework-forger-plugin/src/forger_plugin.ts

```diff
@@ -147,75 +151,85 @@ export class ForgerPlugin extends BasePlugin {
 		await this._forgerPluginDB.close();
 	}
 
-	private _registerMiddlewares(options: Options): void {
-		// Register middlewares
-		this._app.use(cors(options.cors));
-		this._app.use(express.json());
-		this._app.use(rateLimit(options.limits));
-		this._app.use(middlewares.whiteListMiddleware(options));
-	}
+	private async _syncForgerInfo(): Promise<void> {
+		const {
+			header: { height: lastBlockHeight },
+		} = this.codec.decodeBlock(await this._channel.invoke<Buffer>('app:getLastBlock'));
+		const forgerSyncInfo = await getForgerSyncInfo(this._forgerPluginDB);
 
-	private _registerAfterMiddlewares(_options: Options): void {
-		this._app.use(middlewares.errorMiddleware());
-	}
+		if (forgerSyncInfo.syncUptoHeight === lastBlockHeight) {
+			// No need to sync
+			return;
+		}
 
-	private _registerControllers(): void {
-		this._app.get('/v1/hello', controllers.helloController(this._channel));
-	}
+		let syncFromHeight: number;
 
-	// eslint-disable-next-line class-methods-use-this
-	private async _getDBInstance(
-		options: Options,
-		dbName = 'lisk-framework-forger-plugin.db',
-	): Promise<KVStore> {
-		const resolvedPath = options.dataPath.replace('~', os.homedir());
-		const dirPath = path.join(resolvedPath, dbName);
-		await fs.ensureDir(dirPath);
-
-		return new KVStore(dirPath);
+		if (forgerSyncInfo.syncUptoHeight > lastBlockHeight) {
+			// Clear all forging information we have and sync again
+			await this._forgerPluginDB.clear();
+			syncFromHeight = 1;
+		} else {
+			syncFromHeight = lastBlockHeight + 1;
+		}
+
+		let i = syncFromHeight;
+
+		// Sync in batch of 1000 blocks
+		while (i <= lastBlockHeight) {
+			const from = i;
+			const to =
+				from +
+				(from + BLOCKS_BATCH_TO_SYNC < lastBlockHeight
+					? BLOCKS_BATCH_TO_SYNC
+					: lastBlockHeight - from);
+
+			const blocks = await this._channel.invoke<string[]>('app: getBlocksByHeightBetween', {
+				from,
+				to,
+			});
+
+			for (const block of blocks) {
+				await this._incrementForgerInfo(block);
+			}
+
+			i = to;
+		}
+
+		// Update height upto which plugin is synced
+		await setForgerSyncInfo(this._forgerPluginDB, lastBlockHeight);
+
+		// Try to sync again if more blocks forged meanwhile
+		await this._syncForgerInfo();
 	}
 
 	private _subscribeToChannel(): void {
 		// eslint-disable-next-line @typescript-eslint/no-misused-promises
 		this._channel.subscribe('app:block:new', async (eventInfo: EventInfoObject) => {
 			const { block } = eventInfo.data as Data;
+			const {
+				header: { height },
+			} = this._getForgerHeaderAndPayloadInfo(block);
+
 			await this._incrementForgerInfo(block);
+			await setForgerSyncInfo(this._forgerPluginDB, height);
```

Comment:
maybe if we check first we dont need to do previous action? because incase where its out of sync you would delete  the info and sync.. so 
```
await setForgerSyncInfo(this._forgerPluginDB, height);
-- get forger info and check if the block received is updated?
if(!included) {
  await this._incrementForgerInfo(block);
}
```
?

Reply:
Sync info should be updated at the last step, when every other processing is completed.